include/swift/AST/DiagnosticsSIL.def

```diff
@@ -460,12 +463,13 @@ NOTE(autodiff_when_differentiating_function_definition,none,
      "when differentiating this function definition", ())
 NOTE(autodiff_cannot_differentiate_through_inout_arguments,none,
      "cannot differentiate through 'inout' arguments", ())
-NOTE(autodiff_control_flow_not_supported,none,
-     "differentiating control flow is not yet supported", ())
-NOTE(autodiff_loops_not_supported,none,
-     "differentiating loops is not yet supported", ())
 NOTE(autodiff_class_member_not_supported,none,
```

Comment:
We could remove `autodiff_class_member_not_supported` now since it isn't used anymore in the code base

Reply:
This diagnostic (`class_member_not_differentiable`) is actually still relevant:
```
NOTE(autodiff_class_member_not_differentiable,none,
     "member is not differentiable because the corresponding class member "
     "is not '@differentiable'", ())
```

It's triggered in cases like this:
```swift
// Adapted from test/AutoDiff/autodiff_diagnostics.swift:
class Foo : Differentiable {
  // Not marked with `@differentiable`.
  func method2(_ x: Float) -> Float {
    return x
  }
}

@differentiable
func differentiateClassMethod2(x: Float) -> Float {
  // expected-error @+2 {{expression is not differentiable}}
  // expected-note @+1 {{member is not differentiable because the corresponding class member is not '@differentiable'}}
  return Foo().method2(x)
}
```

There's a similar diagnostic for protocol requirements:
```
NOTE(autodiff_protocol_member_not_differentiable,none,
     "member is not differentiable because the corresponding protocol "
     "requirement is not '@differentiable'", ())
```

Reply:
Oops, `autodiff_class_member_not_supported` is the diagnostic we should remove:
```
NOTE(autodiff_class_member_not_supported,none,
     "differentiating class members is not yet supported", ())
```

I'll remove it soon, thanks for the catch 🙂